subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/dependencymanagement/SamplesDeclaringRepositoriesIntegrationTest.groovy

```diff
@@ -17,18 +17,22 @@
 package org.gradle.integtests.samples.dependencymanagement
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.RepoScriptBlockUtil
 import org.gradle.integtests.fixtures.Sample
 import org.gradle.integtests.fixtures.UsesSample
 import org.junit.Rule
 
 class SamplesDeclaringRepositoriesIntegrationTest extends AbstractIntegrationSpec {
 
+    File repositoryMirrorScript = RepoScriptBlockUtil.createMirrorInitScript()
+
     @Rule
     Sample sample = new Sample(testDirectoryProvider)
 
     @UsesSample("userguide/dependencyManagement/declaringRepositories/publicRepository")
     def "can declare JCenter repository and resolve binary dependency"() {
         executer.inDirectory(sample.dir)
+        executer.usingInitScript(repositoryMirrorScript)
```

Comment:
We can probably avoid a couple lines of code if we'd put that into a `setup` block.